eng/scripts/update-version.ps1

```diff
@@ -0,0 +1,144 @@
+#>
+[CmdletBinding()]
+param (
+  [Parameter(Mandatory = $true, HelpMessage = "Absolute path to the SDK repository root")]
+  [string]$SdkRepoPath,
+  
+  [Parameter(Mandatory = $true, HelpMessage = "Absolute path to the SDK package directory")]
+  [string]$PackagePath,
+  
+  [Parameter(Mandatory = $true, HelpMessage = "Release type: 'beta' or 'stable'")]
+  [ValidateSet('beta', 'stable')]
+  [string]$ReleaseType,
+  
+  [Parameter(Mandatory = $true, HelpMessage = "Version number (e.g., '1.0.0' or '1.0.0-beta.1')")]
+  [string]$Version,
+  
+  [Parameter(Mandatory = $true, HelpMessage = "Release date in YYYY-MM-DD format")]
+  [string]$ReleaseDate
+)
+
+# Validates that the path contains a valid package
+function Get-PackageInfo {
+  param (
+    [Parameter(Mandatory = $true)]
+    [string]$PackagePath
+  )
+  
+  try {
+    $resolvedPath = Resolve-Path $PackagePath -ErrorAction Stop
+    $packageJsonPath = Join-Path $resolvedPath "package.json"
+    
+    if (-not (Test-Path $packageJsonPath)) {
+      throw "package.json not found at: $packageJsonPath"
+    }
+    
+    Write-Host "Reading package.json from: $packageJsonPath"
+    
+    $packageJson = Get-Content $packageJsonPath -Raw | ConvertFrom-Json -ErrorAction Stop
+    
+    if (-not $packageJson.name) {
+      throw "'name' field not found in package.json"
+    }
+    
+    return [PSCustomObject]@{
+      Name    = $packageJson.name
+      Path    = $resolvedPath
+      Version = $packageJson.version ?? "unknown"
+      SdkType = $packageJson.'sdk-type' ?? "unknown"
+    }
+  }
+  catch {
+    Write-Host "Error processing package: $($_.Exception.Message)" -ForegroundColor Red
+    throw
+  }
+}
+
+# Main execution
+try {
+  # Validate SDK repository path
+  if (-not (Test-Path $SdkRepoPath)) {
+    throw "SDK repository path does not exist: $SdkRepoPath"
+  }
+  
+  $resolvedRepoPath = Resolve-Path $SdkRepoPath -ErrorAction Stop
+  Write-Host "SDK Repository: $resolvedRepoPath"
+  
+  # Validate and get package information
+  $packageInfo = Get-PackageInfo -PackagePath $PackagePath
+  
+  Write-Host ""
+  Write-Host "Package Information:" -ForegroundColor Cyan
+  Write-Host "  Name:     $($packageInfo.Name)"
+  Write-Host "  Version:  $($packageInfo.Version)"
+  Write-Host "  Type:     $($packageInfo.SdkType)"
+  Write-Host "  Path:     $($packageInfo.Path)"
+  Write-Host ""
+  Write-Host "Version Update Information:" -ForegroundColor Cyan
+  Write-Host "  Release Type: $ReleaseType"
+  Write-Host "  New Version:  $Version"
+  Write-Host "  Release Date: $ReleaseDate"
+  Write-Host ""
+  
+  # Run the update-version command using npx
+  Write-Host "Updating package version..." -ForegroundColor Green
+  Write-Host "Running: npx --yes --package=@azure-tools/js-sdk-release-tools -- update-version --sdkRepoPath `"$resolvedRepoPath`" --packagePath `"$($packageInfo.Path)`" --releaseType `"$ReleaseType`" --version `"$Version`" --releaseDate `"$ReleaseDate`"" -ForegroundColor Gray
```

Comment:
I assume this is for mgmt plane only, but we really should try to start sharing code as we have code to update the version at https://github.com/Azure/azure-sdk-for-js/blob/main/eng/scripts/Language-Settings.ps1#L386 also

Reply:
+1 we should see whether the script works for mgmt-plane SDK case.